scripts/install-step-ra.sh

```diff
@@ -0,0 +1,253 @@
+#!/bin/bash
+set -e
+
+echo "This script will install and start a step-ca server running in Registration Authority (RA) mode."
+echo ""
+echo "You will need an upstream CA (URL and fingerprint)"
+echo "Don't have a CA? Sign up for a hosted CA at smallstep.com — or run your own."
+echo ""
+
+# Fail if this script is not run as root.
+if ! [ $(id -u) = 0 ]; then
+  echo "This script must be run as root"
+  exit 1
+fi
+
+# Architecture detection
+arch=$(uname -m)
+case $arch in
+  x86_64) arch="amd64" ;;
+  x86) arch="386" ;;
+  i686) arch="386" ;;
+  i386) arch="386" ;;
+  aarch64) arch="arm64" ;;
+  armv5*) arch="armv5" ;;
+  armv6*) arch="armv6" ;;
+  armv7*) arch="armv7" ;;
+esac
+
+if [ "$arch" = "armv5" ]; then
+  echo "This script doesn't work on armv5 machines"
+  exit 1
+fi
+
+if ! hash jq &> /dev/null; then
+  echo "This script requires the jq commmand; please install it."
+  exit 1
+fi
+
+if ! hash curl &> /dev/null; then
+  echo "This script requires the curl commmand; please install it."
+  exit 1
+fi
+
+if ! hash tar &> /dev/null; then
+  echo "This script requires the tar commmand; please install it."
+  exit 1
+fi
+
+while [ $# -gt 0 ]; do
+  case "$1" in
+    --ca-url)
+      CA_URL="$2"
+      shift
+      shift
+      ;;
+    --fingerprint)
+      CA_FINGERPRINT="$2"
+      shift
+      shift
+      ;;
+    --provisioner-name)
+      CA_PROVISIONER_NAME="$2"
+      shift
+      shift
+      ;;
+    --provisioner-password-file)
+      CA_PROVISIONER_JWK_PASSWORD_FILE="$2"
+      shift
+      shift
+      ;;
+    --dns-names)
+      RA_DNS_NAMES="$2"
+      shift
+      shift
+      ;;
+    --listen-address)
+      RA_ADDRESS="$2"
+      shift
+      shift
+      ;;
+    *)
+      shift
+      ;;
+  esac
+done
+
+# Install step
+if ! hash step &> /dev/null; then
+  echo "Installing 'step' in /usr/bin..."
+  STEP_VERSION=$(curl -s https://api.github.com/repos/smallstep/cli/releases/latest | jq -r '.tag_name')
+
+  curl -sLO https://github.com/smallstep/cli/releases/download/$STEP_VERSION/step_linux_${STEP_VERSION:1}_$arch.tar.gz
+  tar xvzf step_linux_${STEP_VERSION:1}_$arch.tar.gz
+  install -m 0755 -t /usr/bin step_${STEP_VERSION:1}/bin/step
```

Comment:
I'm more of a `/usr/local/bin` guy for those packages that are not installed using a package manager.

Reply:
I agree. @dopey and I had to make a call at some point, about where `step-ca` should live, because the path is hardcoded into the systemd unit and in one or two places in the docs. Maybe there's a way around this? I used to use `/usr/local/bin`, but since the Debian package installs it in `/usr/bin` we started using that everywhere. :(